components/metrics/http_test.go

```diff
@@ -49,3 +50,16 @@ func TestCreateRegistryAndServeHTTP_unknown_endpoint(t *testing.T) {
 	assert.NotNil(t, resp)
 	assert.Equal(t, http.StatusNotFound, resp.StatusCode)
 }
+
+// server might have small delay before being able to server traffic
+func waitServerReady(t *testing.T, addr string) {
+	for i := 0; i < 50; i++ {
+		_, err := http.DefaultClient.Get(addr)
```

Comment:
I'd suggest not using the default client because he has no timeouts set https://golang.org/src/net/http/client.go line 110

Reply:
Thanks for the feedback, it was a while and I forgot a bit about this PR.
For the default http client, I agree with you and I would not use it for production code. However since we are in tests, I was considering leaving the default test timeout handling this instead, I found it simpler. What do you think?

Reply:
ok, tests have their timeouts so it can stay as it is